src/algebraic_geometry/structured_stalks.lean

```diff
@@ -0,0 +1,95 @@
+-- Copyright (c) 2019 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison, Mario Carneiro, Reid Barton
+
+import algebraic_geometry.stalks
+
+universes v₁ v₂ u₁ u₂
+
+open category_theory
+open category_theory.limits
+
+namespace algebraic_geometry
+
+variables (C : Type u₁) [𝒞 : category.{v₁+1} C]
+variables (V : Type u₂) [𝒱 : category.{v₂+1} V]
+include 𝒞 𝒱
+variables [has_colimits.{v₁} C]
+
+structure StructuredStalkPresheafedSpace (F : V ⥤ C) extends PresheafedSpace.{v₁} C :=
+(structured_stalk : Π x : to_Top, V)
+(compatible : Π x : to_Top, F.obj (structured_stalk x) ≅ (to_PresheafedSpace.stalk x))
+
+namespace StructuredStalkPresheafedSpace
+variables {C V}
+variables {F : V ⥤ C}
+
+instance : has_coe (StructuredStalkPresheafedSpace.{v₁ v₂} C V F) (PresheafedSpace.{v₁} C) :=
+{ coe := λ X, X.to_PresheafedSpace }
+
+structure hom (X Y : StructuredStalkPresheafedSpace.{v₁ v₂} C V F) :=
+(hom : X.to_PresheafedSpace ⟶ Y.to_PresheafedSpace)
+(structured_stalk_map :
+  Π (x : X.to_Top), Y.structured_stalk ((hom : X.to_Top → Y.to_Top) x) ⟶ X.structured_stalk x)
+(compatible' : Π (x : X.to_Top),
+  F.map (structured_stalk_map x) =
+    (Y.compatible (hom x)).hom ≫ PresheafedSpace.stalk_map hom x ≫ (X.compatible x).inv . obviously)
+
+restate_axiom hom.compatible'
+attribute [simp] hom.compatible
+
+@[extensionality] lemma hom.ext
+  {X Y : StructuredStalkPresheafedSpace.{v₁ v₂} C V F} {f g : hom X Y}
+  (w : f.hom = g.hom)
+  (h : ∀ x, f.structured_stalk_map x = by { convert g.structured_stalk_map x, rw w }) : f = g :=
```

Comment:
I personally think that heterogeneous equality would be more convenient here:
```
(h : ∀ x, f.structured_stalk_map x == g.structured_stalk_map x)
```
But I've seen this rewriting in the type of a lemma before in the library, so feel free to ignore this comment (I don't know how well common tactics deal with heterogeneous equality).

Reply:
I've always steered clear of heterogeneous equality; seems safer. :-) I'd prefer to ignore this for now unless we see a compelling reason.